ddl/check.lua

```diff
@@ -628,75 +628,131 @@ local function check_index(i, index, space)
     return true
 end
 
-local function check_sharding_key(space)
-    if not space.sharding_key then
-        if space.fields.bucket_id ~= nil then
-            return nil, string.format(
-                "spaces[%q].format[%q]: bucket_id is used for sharding, " ..
-                "but there's no spaces[%q].sharding_key defined",
-                space.name, 'bucket_id', space.name
-            )
-        end
+local function check_bucket_id_field_value(space)
+    if not space.fields.bucket_id then
+        return nil, string.format(
+            "spaces[%q].format: sharding_key exists in the space, but there's" ..
+            " no bucket_id defined in 'format' section",
+            space.name
+        )
+    end
 
-        if space.indexes.bucket_id ~= nil then
-            return nil, string.format(
-                "spaces[%q].indexes[%q]: bucket_id is used for sharding, " ..
-                "but there's no spaces[%q].sharding_key defined",
-                space.name, 'bucket_id', space.name
-            )
-        end
+    if space.fields.bucket_id.type ~= 'unsigned' then
+        return nil, string.format(
+            "spaces[%q].format[%q].type: bad value (unsigned expected, got %s)",
+            space.name, 'bucket_id', space.fields.bucket_id.type
+        )
+    end
+
+    return true
+end
+
+local function check_bucket_id_index(space)
+    if not space.indexes.bucket_id then
+        return nil, string.format(
+            "spaces[%q].indexes: sharding_key exists in the space, but there's" ..
+            " no bucket_id defined in 'indexes' section",
+            space.name
+        )
+    end
+
+    if space.indexes.bucket_id.unique then
+        return nil, string.format(
+            "spaces[%q].indexes[%q].unique: bucket_id index can't be unique",
+            space.name, 'bucket_id'
+        )
+    end
+
+    if #space.indexes['bucket_id'].parts ~= 1 then
+        return nil, string.format(
+            "spaces[%q].indexes[%q].parts: bucket_id index can't be composite (1 part expected, got %d parts)",
+            space.name, 'bucket_id', #space.indexes['bucket_id'].parts
+        )
+    end
+
+    if space.indexes['bucket_id'].parts[1].path ~= 'bucket_id' then
+        return nil, string.format(
+            "spaces[%q].indexes[%q].parts[1].path: invalid field reference " ..
+            "(reference to bucket_id expected, got %s)",
+            space.name, 'bucket_id', space.indexes['bucket_id'].parts[1].path
+        )
+    end
+
+    return true
+end
+
+local function check_custom_sharding_func_is_valid(sharding_func_name)
```

Comment:
All functions in this module check data for validness. But don't use postfix `_is_valid` in name. I think we can follow this rules here too.

Reply:
Updated.